sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -308,6 +314,19 @@ public Double getFractionConsumed() {
     }
 
     @Override
+    public long getParallelismConsumed() {
+      return rangeTracker.getProcessedSplitPoints();
+    }
+
+    @Override
+    public long getParallelismRemaining() {
+      if (getCurrentOffset() + 1 >= rangeTracker.getStopPosition()) {
```

Comment:
Why + 1? Typically, if getCurrentOffset == getStopPosition, we would be done (0), so the fact that we are not done would indicate already 1. This source could also track done to return 0.